### PR TITLE
Finish cleaning up directory references

### PR DESCRIPTION
**Pull request by madhurajayaraman**
Test-On-Device: true

b/302730696

### CODE REVIEW

File: starboard/shared/win32/posix_emu/dirent.cc [hunk 1]
```diff
@@ -180,18 +190,26 @@ DIR* opendir(const char* path) {
...
+  if (!dir) {
+    errno = EBADF;
+    return -1;
+  }
   bool success = CloseHandle(dir->handle);
   handle_db_get(dir->fd, true);
   delete dir;
-  return success ? 0 : -1;
+  if (!success) {
+    errno = EINTR;
```
Comment (kaidokert) [inline starboard/shared/win32/posix_emu/dirent.cc:201]: This one looks a little random - is it supposed to be EINTR or something more fatal ?

Comment (madhurajayaraman) [reply starboard/shared/win32/posix_emu/dirent.cc:201]: https://pubs.opengroup.org/onlinepubs/7908799/xsh/errors.html 
I couldn't figure out what else could work instead, there's not really a catch all fatal error that I could find

File: starboard/shared/win32/posix_emu/dirent.cc [hunk 2]
```diff
@@ -170,7 +179,8 @@ DIR* opendir(const char* path) {
   if (!basic_info_success ||
       !(basic_info.FileAttributes & FILE_ATTRIBUTE_DIRECTORY)) {
     CloseHandle(directory_handle);
-    return NULL;
+    errno = ENOTDIR;
+    return nullptr;
   }
 
   DIR* dir = reinterpret_cast<DIR*>(malloc(sizeof(DIR)));
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply starboard/shared/win32/posix_emu/dirent.cc]: Changed.

File: starboard/shared/win32/posix_emu/dirent.cc [hunk 3]
```diff
@@ -180,18 +190,26 @@ DIR* opendir(const char* path) {
 }
 
 int closedir(DIR* dir) {
+  if (!dir) {
+    errno = EBADF;
+    return -1;
+  }
   bool success = CloseHandle(dir->handle);
   handle_db_get(dir->fd, true);
   delete dir;
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply starboard/shared/win32/posix_emu/dirent.cc]: We used new for Sbxxx allocation, and I missed modifying this one when copying over. Thanks for catching it!

**Review by y4vor — APPROVED** [review]

**Pull request merged**